src/app/beer_garden/api/http/handlers/v1/user.py

```diff
@@ -123,93 +91,15 @@ def patch(self, user_id):
         tags:
           - Users
         """
-        principal = Principal.objects.get(id=str(user_id))
-        operations = MongoParser.parse_patch(
-            self.request.decoded_body, many=True, from_string=True
-        )
-
-        # Most things only need a permission check if updating a different user
-        if user_id != str(self.current_user.id):
-            check_permission(self.current_user, [Permissions.USER_UPDATE])
-
-        for op in operations:
-            if op.path == "/roles":
-                # Updating roles always requires USER_UPDATE
-                check_permission(self.current_user, [Permissions.USER_UPDATE])
-
-                try:
-                    if op.operation == "add":
-                        principal.roles.append(LegacyRole.objects.get(name=op.value))
-                    elif op.operation == "remove":
-                        principal.roles.remove(LegacyRole.objects.get(name=op.value))
-                    elif op.operation == "set":
-                        principal.roles = [
-                            LegacyRole.objects.get(name=name) for name in op.value
-                        ]
-                    else:
-                        raise ModelValidationError(
-                            "Unsupported operation '%s'" % op.operation
-                        )
-                except DoesNotExist:
-                    raise ModelValidationError("Role '%s' does not exist" % op.value)
-
-            elif op.path == "/username":
-
-                if op.operation == "update":
-                    principal.username = op.value
-                else:
-                    raise ModelValidationError(
-                        "Unsupported operation '%s'" % op.operation
-                    )
-
-            elif op.path == "/password":
-                if op.operation != "update":
-                    raise ModelValidationError(
-                        "Unsupported operation '%s'" % op.operation
-                    )
-
-                if isinstance(op.value, dict):
-                    current_password = op.value.get("current_password")
-                    new_password = op.value.get("new_password")
-                else:
-                    current_password = None
-                    new_password = op.value
+        user_data = UserCreateSchema().load(self.request_body, partial=True).data
```

Comment:
Empty dictionary behaves as intended, which is that nothing gets updated and you get a 200 response saying everything is okay. I guess I could see an argument for assuming that an empty dictionary is most likely not what someone intended and throwing back a 400 (Bad Request) would be more useful to them in that case, but I don't know if I like that.

`None` on the other hand does not behave the way it should. I was thinking that there's no way to get here with `request_body` being `None`, but that's not true since someone could send a PATCH with no body at all.  Right now that results in this code blowing up and a 500 response being returned, which is definitely not correct.

What I think should happen is that instead of `request_body` ever return `None`, it should just raise some exception that gets handled to respond with a `400`, since not sending a body for your PATCH would very much meet the definition of a "Bad Request".